components/base/kinematicbase/kinematics.go

```diff
@@ -202,3 +222,45 @@ func CollisionGeometry(cfg *referenceframe.LinkConfig) ([]spatialmath.Geometry,
 	}
 	return []spatialmath.Geometry{sphere}, nil
 }
+
+// newValidRegionCapsule returns a capsule which defines the valid regions for a base to be when moving to a waypoint.
+// The valid region is all points that are deviationThreshold (mm) distance away from the line segment between the
+// starting and ending waypoints. This capsule is used to detect whether a base leaves this region and has thus deviated
+// too far from its path.
+func (ddk *differentialDriveKinematics) newValidRegionCapsule(starting, desired []referenceframe.Input) (spatialmath.Geometry, error) {
+	pt := r3.Vector{X: (desired[0].Value + starting[0].Value) / 2, Y: (desired[1].Value + starting[1].Value) / 2}
+	positionErr, _, err := ddk.errorState(starting, desired)
+	if err != nil {
+		return nil, err
+	}
+
+	// rotate such that y is forward direction to match the frame for movement of a base
+	axesRotation := spatialmath.NewPoseFromOrientation(&spatialmath.R4AA{
+		Theta: math.Pi / 2,
+		RX:    1,
+		RY:    0,
+		RZ:    0,
+	})
+
+	// rotate around the z-axis such that the capsule points in the direction of the end waypoint
+	desiredHeading := math.Atan2(starting[0].Value-desired[0].Value, starting[1].Value-desired[1].Value)
+	headingRotation := spatialmath.NewPoseFromOrientation(&spatialmath.R4AA{
+		Theta: -desiredHeading,
+		RX:    0,
+		RY:    0,
+		RZ:    1,
+	})
+
+	capsule, err := spatialmath.NewCapsule(
+		spatialmath.Compose(headingRotation, axesRotation),
+		deviationThreshold,
+		2*deviationThreshold+float64(positionErr),
+		"")
+	if err != nil {
+		return nil, err
+	}
+
+	transformedCapsule := capsule.Transform(spatialmath.NewPoseFromPoint(pt))
```

Comment:
We talked offline but leaving a comment here to try to collapse these transformations into a pose that we pass arguments to in order to construct